src/ucs/sys/sock.c

```diff
@@ -405,10 +405,12 @@ ucs_socket_do_iov_nb(int fd, struct iovec *iov, size_t iov_cnt, size_t *length_p
     if (ucs_likely(/* The IO operation is considered as successful if: */
                    /* - the return value > 0 */
                    (ret > 0) ||
-                   /* - the number of user's IOV elements == 0 and the
-                    *   retrun value == 0 (Datagram sockets permit
-                    *   zero-length datagrams) */
-                   ((iov_cnt == 0)  && (ret == 0)))) {
+                   /* - the return value == 0 and the user's data length == 0
+                    *   (the number of the iov array buffers == 0 or the total
+                    *   length of the iov array buffer == 0) */
+                   ((ret == 0) &&
+                    ((iov_cnt == 0) ||
+                     (ucs_iov_total_length(iov, iov_cnt) == 0))))) {
```

Comment:
can we unit common code to a function which checks if "the IO operation is considered as successful"
function parameters: ret, length

Reply:
done